pkg/context/http/context.go

```diff
@@ -36,7 +38,7 @@ type HttpContext struct {
 	HttpConnectionManager model.HttpConnectionManager
 	FilterChains          []model.FilterChain
 	Listener              *model.Listener
-	api                   *model.Api
+	api                   router.API
```

Comment:
why not pointer?

Reply:
i don't want the origin api be changed in the context. So I change it to not use the pointer